src/utils/featheryClient/index.ts

```diff
@@ -574,11 +574,27 @@ export default class FeatheryClient extends IntegrationClient {
       data.field_values = fieldValues;
     }
 
-    return this._fetch(`${API_URL}custom_request/`, {
+    const url = `${API_URL}custom_request/`;
+    const options = {
       headers: { 'Content-Type': 'application/json' },
       method: 'POST',
       body: JSON.stringify(data)
-    }).then((response) => (response ? response.json() : Promise.resolve()));
+    };
+    if (
+      typeof payload !== 'string' &&
```

Comment:
why do we check the payload type here?

Reply:
payload could be a string and payload.method will give TS error